### PR TITLE
feat(open-pr-comments): python patch parser

### PR DESCRIPTION
**Pull request by cathteng**
Use a regex to parse Python function names from git diff. Make this a class so that new parsers can be easily added -- we can fetch the correct parser according to a file's suffix.

For ER-1805

### CODE REVIEW

File: tests/sentry/integrations/utils/test_patch_parsers.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
+from sentry.integrations.utils.patch_parsers import PythonPatch
+from sentry.testutils.cases import TestCase
+
+
+class PythonPatchTestCase(TestCase):
```
Comment (dashed) [inline tests/sentry/integrations/utils/test_patch_parsers.py:5]: A test case for inputs that are not patches (e.g. malicious inputs) would be good to have.

Comment (cathteng) [reply tests/sentry/integrations/utils/test_patch_parsers.py:5]: what would be a malicious input in this case? do you mean inputs that don't match the regex at all?

File: tests/sentry/integrations/utils/test_patch_parsers.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
++    def create_org_auth_token(self, *args, **kwargs):
++        return Factories.create_org_auth_token(*args, **kwargs)
++
+     def create_sentry_app_installation(self, *args, **kwargs):
+         return Factories.create_sentry_app_installation(*args, **kwargs)"""
+        functions = PythonPatch().extract_from_patch(patch=patch)
+
+        assert len(functions) == 1
+        assert "create_member" in functions
```
[the remark this answers is not in the document]
Comment (EricHasegawa) [reply tests/sentry/integrations/utils/test_patch_parsers.py:56]: Maybe I'm misunderstanding the scope of this - feel free to correct me if these aren't relevant

File: tests/sentry/integrations/utils/test_patch_parsers.py (hunk 1 above)
Comment (dashed) [reply tests/sentry/integrations/utils/test_patch_parsers.py:5]: @cathteng Yea, strings that won't match the regex is that I was thinking of.

**Pull request closed without merging**